components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1354,6 +1322,60 @@ private static String getTenantUrl(String url, String tenantDomain) throws URISy
         }
     }
 
+    /**
+     * Builds a URL with a given context in both the tenant-qualified url supported mode and the legacy mode. Returns
+     * the absolute URL build from the default context in the tenant-qualified url supported mode and returns the
+     * absolute url build from file configuration context.
+     *
+     * @param defaultContext              Default URL context.
+     * @param getValueFromFileBasedConfig File-based Configuration.
+     * @return Absolute URL.
+     */
+    public static String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig) {
```

Comment:
Should this be public ?

Reply:
Yes, we need to call this in the ProviderConfigBuilder